charts/opentelemetry-operator/values.schema.json

```diff
@@ -1336,12 +1336,31 @@
                     }]
                 },
                 "autoGenerateCert": {
-                    "type": "boolean",
-                    "default": true,
-                    "title": "The autoGenerateCert schema",
-                    "examples": [
-                        true
-                    ]
+                    "type": "object",
```

Comment:
Have you confirmed that this will cause users who are doing `autoGenerateCert: false` to fail an install?

Reply:
Yes, it fails with an error - invalid type, expected object instead of boolean.
I can try to use oneOf in schema definition and detect which version is provided, what do you think?

Reply:
What you've done here is correct, but we should add an entry to UPGRADING.md that explains how to handle this breaking change since users doing `autoGenerateCert: false` will not be able to upgrade until that update their values.yaml.

Reply:
The same situation is with `autoGenerateCert: true`, it looks like a breaking change for me.

Reply:
I've updated UPGRADING.md

Reply:
@tomplus true.  If the user explicitly set the value to `autoGenerateCert: true` (which would be unnecessary since it is the default value), they'd experience an error.